### PR TITLE
Fix service factory defined in yaml configuration

### PR DESCRIPTION
**Pull request by glickel**

### CODE REVIEW

**Comment by wouterj** [comment]
Hi @glickel! Thank you for creating a pull request.

As it confused me why we had one colon here in the first place, I digged a bit into the code. The relevant part of [`YamlFileLoader::parseCallable()`](https://github.com/symfony/dependency-injection/blob/f44572c0fa94ad4fd3c7a14286cbd48a65cb9968/Loader/YamlFileLoader.php#L586-L618) looks like this:

```php
        if (\is_string($callable)) {
            if ('' !== $callable && '@' === $callable[0]) {
                if (false === strpos($callable, ':')) {
                    return [$this->resolveServices($callable, $file), '__invoke'];
                }
                throw new InvalidArgumentException(sprintf('The value of the "%s" option for the "%s" service must be the id of the service without the "@" prefix (replace "%s" with "%s").', $parameter, $id, $callable, substr($callable, 1)));
            }

            if (false !== strpos($callable, ':') && false === strpos($callable, '::')) {
                $parts = explode(':', $callable);

                return [$this->resolveServices('@'.$parts[0], $file), $parts[1]];
            }

            return $callable;
        }
```

As you can see, with one colon, the first part is parsed as a service name. With a double colon, the string is just returned as callable.

This means `NewsletterManagerFactory:createNewsletterManager` is first initializing the `NewsletterManagerFactory` service and then calling `createNewsletterManager` on it. In plain PHP, it's like:

```php
$factory = new NewsletterManagerFactory();
$factory->createNewsletterManager();
```

`NewsletterManagerFactory::createNewsletterManager` on the other hand is just the PHP callable string. Resulting in this PHP code:

```php
NewsletterManagerFactory::createNewsletterManager();
```

In other words, a single or double colon have a different meaning. And now we're using class names as service names, it's getting quite confusing. In this location, we need the single colon class as we need to use the `AppBundle\Email\NewsletterManagerFactory` _service_.

I hope to have explained it correctly and if you have suggestions to make this more clear in the documentation, feel free to welcome! Unfortunately, it does mean I have to close this PR. Hope to see you back in the future with another contribution!
